// Protocol message to describe model checkpoints.
//
// Copyright 2019-2020 the ProGraML authors.
//
// Contact Chris Cummins <chrisc.101@gmail.com>.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package programl;

option go_package = "programlpb";
option java_multiple_files = true;
option java_outer_classname = "CheckpointProto";
option java_package = "com.programl";

// A model checkpoint.
//
// A checkpoint captures a snapshot of a model's state. The model data is stored
// as an opaque binary blob - it is up to the implementation to serialize and
// deserialize state from this byte array.
message Checkpoint {
  bytes model_data = 2;
  reserved 1, 3, 4; // Deprecated tags.
}
